### PR TITLE
SK personal code and ID card number generators

### PR DESCRIPTION
**Pull request by mikk150**

### CODE REVIEW

**Comment by mikk150** [comment]
DO NOT MERGE RIGHT NOW! FOUND BUG!

File: src/Faker/Provider/sk_SK/Person.php [hunk 1]
```diff
@@ -165,4 +165,46 @@ public static function suffix()
...
+     *
+     * @link https://en.wikipedia.org/wiki/National_identification_number#Slovakia
+     * @param  \DateTime|null $birthDate date when person to be generated was born
+     * @param  string $gender person gender
+     * @return string generated birth number
+     */
+    public static function birthNumber($birthDate = null, $gender = null)
+    {
+        $birthNumber='';
+        $gender = $gender?$gender:\Faker\Provider\Base::randomElement(array(static::GENDER_MALE, static::GENDER_FEMALE));
```
Comment (fzaninotto) [inline src/Faker/Provider/sk_SK/Person.php]: don't make the method `static`, and use `$this->randomElement()` instead.

also, coding standards are wrong, spaces are missing. See the entire codebase for examples.

File: src/Faker/Provider/sk_SK/Person.php [hunk 2]
```diff
@@ -165,4 +165,46 @@ public static function suffix()
...
+        /**
+         * @var \DateTime
+         */
+        $birthDate = $birthDate?$birthDate:\Faker\Provider\DateTime::dateTime();
```
Comment (fzaninotto) [inline src/Faker/Provider/sk_SK/Person.php]: same, use `static` or `$this`

File: src/Faker/Provider/sk_SK/Person.php [hunk 3]
```diff
@@ -165,4 +165,46 @@ public static function suffix()
...
+
+        $birthNumber .= $birthDate->format('y');
+        $birthNumber .= $gender==static::GENDER_MALE? $birthDate->format('m'): ($birthDate->format('m')+50); // females are indicated by birth date month number +50
+        $birthNumber .= $birthDate->format('d');
+
+        $padding = \Faker\Provider\Base::numberBetween(100, 999);
```
Comment (fzaninotto) [inline src/Faker/Provider/sk_SK/Person.php]: same

File: src/Faker/Provider/sk_SK/Person.php [hunk 4]
```diff
@@ -165,4 +165,46 @@ public static function suffix()
...
+
+        if ($birthDate->format('U')<-504921600) { // if born before 1954-01-01 then there is no check digit
```
Comment (fzaninotto) [inline src/Faker/Provider/sk_SK/Person.php]: use a constant value instead of a number

File: src/Faker/Provider/sk_SK/Person.php [hunk 5]
```diff
@@ -165,4 +165,46 @@ public static function suffix()
...
+
+    /**
+     * @example RY432655
+     *
+     * @link https://en.wikipedia.org/wiki/National_identification_number#Slovakia
+     * @return string generated IdCard number
+     */
+    public static function idCard()
+    {
+        return \Faker\Provider\Base::regexify('[A-Z]{2}[0-9]{6}');
```
Comment (fzaninotto) [inline src/Faker/Provider/sk_SK/Person.php]: use `static` instead

**Comment by fzaninotto** [comment]
Please look at other providers for coding standards and the right way to reuse existing formatters.

Also, you must add a unit test for the new formatters, and a mention in the locale-specific section of the README

**Comment by fzaninotto** [comment]
Bump, see my previous comment

**Comment by localheinz** [comment]
@mikk150 

Are you still interested in working on this? If so, we could help you get this through!

**Comment by pimjansen** [comment]
Going to close this issue since it is stale. If still needed feel free to reopen it again

**Pull request closed without merging**